src/ResourceManagement/Compute/EncryptionSettings.cs

```diff
@@ -0,0 +1,187 @@
+        {
+            return new EncryptionSettings.Enable<T>(config);
+        }
+
+
+        /// <summary>
+        /// Internal type exposing settings for disabling disk encryption.
+        /// </summary>
+        public partial class Disable : EncryptionSettings
+        {
+
+            private readonly DiskVolumeType volumeType;
+
+            internal Disable(DiskVolumeType volumeType)
+            {
+                this.volumeType = volumeType;
+            }
+
+            internal override IDictionary<string, object> ExtensionProtectedSettings()
+            {
+                return new Dictionary<string, object>();
+            }
+
+            internal override IDictionary<string, object> ExtensionPublicSettings()
+            {
+                Dictionary<string, object> publicSettings = new Dictionary<string, object>();
+                publicSettings.Add("EncryptionOperation", "DisableEncryption");
+                publicSettings.Add("SequenceVersion", Guid.NewGuid());
+                publicSettings.Add("VolumeType", this.volumeType);
+                return publicSettings;
+            }
+
+            internal override DiskEncryptionSettings StorageProfileEncryptionSettings()
+            {
+                DiskEncryptionSettings diskEncryptionSettings = new DiskEncryptionSettings();
+                diskEncryptionSettings.Enabled = false;
+                return diskEncryptionSettings;
+            }
+        }
+
+        /// <summary>
+        /// Internal type exposing settings for enabling disk encryption.
+        /// </summary>
+        /// <param>T</param>
+        public partial class Enable<T> : EncryptionSettings
+            where T : VirtualMachineEncryptionConfiguration<T>
+        {
+            private readonly VirtualMachineEncryptionConfiguration<T> config;
+
+            internal Enable(VirtualMachineEncryptionConfiguration<T> config)
+            {
+                this.config = config;
+
+            }
+
+            /// <return>The encryption version based on user selected OS and encryption extension.</return>
+            internal string EncryptionExtensionVersion()
+            {
+                return EncryptionExtensionIdentifier.Version(this.config.OsType(), RequestedForNoAADEncryptExtension());
+            }
+
+            internal override IDictionary<string, object> ExtensionProtectedSettings()
+            {
+                if (this.RequestedForLegacyEncryptExtension())
+                {
+                    Dictionary<string, object> protectedSettings = new Dictionary<string, object>();
+                    // Legacy-Encrypt-Extension requires AAD credentials (AADClientID in PublicSettings & AADClientSecret in ProtectedSettings) to access KeyVault.
+                    protectedSettings.Add("AADClientSecret", config.AadSecret());
+                    if (config.OsType() == OperatingSystemTypes.Linux
+                    && config.LinuxPassPhrase() != null)
```

Comment:
How about Windows system behavior?

Reply:
windows does not have such field, see https://docs.microsoft.com/en-us/azure/virtual-machines/extensions/azure-disk-enc-windows#schema-v01-with-aad and https://docs.microsoft.com/en-us/azure/virtual-machines/extensions/azure-disk-enc-linux#schema-v01-with-aad